### PR TITLE
Make Plank use a caching PJ client.

### PR DESCRIPTION
**Pull request by cjwagner**
We're seeing some issues with the API server taking too long to list PJs this is a mitigation for the most critical components.
/assign @chases2 @Katharine @stevekuznetsov 

I couldn't figure out what the default value of the timeout is from the docs and gave up trying to find it in the code.

### CODE REVIEW

File: prow/cmd/sinker/main.go [hunk 1]
```diff
@@ -297,7 +297,18 @@ func (c *controller) clean() {
 
 	// Clean up old prow jobs first.
 	prowJobs := &prowapi.ProwJobList{}
-	if err := c.prowJobClient.List(c.ctx, prowJobs, ctrlruntimeclient.InNamespace(c.config().ProwJobNamespace)); err != nil {
+	timeoutSecs := (*int64)(nil)
+	if dur := c.config().Sinker.ProwJobListTimeout; dur != nil {
+		val := int64(dur.Duration.Seconds())
+		timeoutSecs = &val
+	}
+	listOpts := &ctrlruntimeclient.ListOptions{
```
Comment (alvaroaleman) [inline prow/cmd/sinker/main.go]: We use a cache-backed client for sinker, this is effectively a timeout for copying the objects in memory

File: prow/plank/controller.go [hunk 2]
```diff
@@ -200,9 +200,17 @@ func (c *Controller) Sync() error {
 	var syncErrs []error
 
 	pjs := &prowapi.ProwJobList{}
+	timeoutSecs := (*int64)(nil)
+	if dur := c.config().Plank.ProwJobListTimeout; dur != nil {
```
Comment (alvaroaleman) [inline prow/plank/controller.go]: Maybe just change to also use a cache-backed client so we reduce the total  number of lists calls needed?

**Comment by cjwagner** [comment]
> > We're seeing some issues with the API server taking too long to list PJs this is a mitigation for the most critical components.
> 
> Can you elaborate? Where do you see that, what impact does it have?

We've been seeing errors like this: `error listing prow jobs: the server was unable to return a response in the time allotted, but may still be processing the request (get prowjobs.prow.k8s.io)` on an internal Prow cluster with fewer ProwJob CRDs than k8s-prow. This prevents plank from syncing any ProwJobs. We're looking into the API server slowness as well, but would like to mitigate in the meantime.

File: prow/cmd/sinker/main.go (hunk 1 above)
Comment (cjwagner) [reply prow/cmd/sinker/main.go]: ACK, updated to only change plank.

File: prow/plank/controller.go (hunk 2 above)
Comment (cjwagner) [reply prow/plank/controller.go]: Done.

**Comment by stevekuznetsov** [comment]
/hold cancel

Would love to see APIServer request latency and size metrics for these calls.

**Pull request merged**